Scripts/Editor/Build/AndroidKeystoreHelper.cs

```diff
@@ -0,0 +1,208 @@
+using Anvil.CSharp.Logging;
+using Anvil.Unity.Editor.UI;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Anvil.Unity.Editor.Build
+{
+
+#if UNITY_ANDROID
+    /// <summary>
+    /// Caches keystore information to <see cref="EditorPrefs"/> to that developers don't have to re-enter it each time
+    /// Unity is launched.
+    /// </summary>
+    public class AndroidKeystoreHelper : IPreprocessBuildWithReport
+    {
+        private const string MENU_PATH_PREFIX = "Anvil/Keystore";
+        private static readonly Logger Logger = Log.GetStaticLogger(typeof(AndroidKeystoreHelper));
+
+        private static string PrefsKeyPrefix
+        {
+            get => $"{MENU_PATH_PREFIX}:{PlayerSettings.productName}-{nameof(AndroidKeystoreHelper)}";
+        }
+        private static string KeystorePassPrefsKey
+        {
+            get => $"{PrefsKeyPrefix}-keystorePass";
+        }
+        private static string KeystoreAliasPassPrefsKey
+        {
+            get => $"{PrefsKeyPrefix}-keystoreAliasPass";
+        }
+
+        /// <summary>
+        /// The keystore values that Unity doesn't clear between sessions.
+        /// </summary>
+        private static bool ArePersistentKeystoreValuesSet
+        {
+            get => !string.IsNullOrEmpty(PlayerSettings.Android.keystoreName)
+                   && !string.IsNullOrEmpty(PlayerSettings.Android.keyaliasName);
+        }
+
+        private static bool AreKeystorePasswordsSet
+        {
+            get => !string.IsNullOrEmpty(PlayerSettings.Android.keystorePass)
+                   && !string.IsNullOrEmpty(PlayerSettings.Android.keyaliasPass);
+        }
+
+
+        [MenuItem(MENU_PATH_PREFIX+"/Set Keystore Passwords")]
+        private static void SetKeystorePasswords()
```

Comment:
I initially thought this meant "set keystore passwords in player settings", but it means "set keystore passwords stored in player prefs"—not a terribly important distinction but maybe calling it "Set Stored Keystore Passwords" or similar would be clearer?

Reply:
Good point, changed to "Set Saved Keystore Passwords"